torchvision/prototype/datasets/_builtin/cityscapes.py

```diff
@@ -0,0 +1,201 @@
+            "vegetation": CityscapesClass("vegetation", 21, 8, "nature", 4, False, False, (107, 142, 35)),
+            "terrain": CityscapesClass("terrain", 22, 9, "nature", 4, False, False, (152, 251, 152)),
+            "sky": CityscapesClass("sky", 23, 10, "sky", 5, False, False, (70, 130, 180)),
+            "person": CityscapesClass("person", 24, 11, "human", 6, True, False, (220, 20, 60)),
+            "rider": CityscapesClass("rider", 25, 12, "human", 6, True, False, (255, 0, 0)),
+            "car": CityscapesClass("car", 26, 13, "vehicle", 7, True, False, (0, 0, 142)),
+            "truck": CityscapesClass("truck", 27, 14, "vehicle", 7, True, False, (0, 0, 70)),
+            "bus": CityscapesClass("bus", 28, 15, "vehicle", 7, True, False, (0, 60, 100)),
+            "caravan": CityscapesClass("caravan", 29, 255, "vehicle", 7, True, True, (0, 0, 90)),
+            "trailer": CityscapesClass("trailer", 30, 255, "vehicle", 7, True, True, (0, 0, 110)),
+            "train": CityscapesClass("train", 31, 16, "vehicle", 7, True, False, (0, 80, 100)),
+            "motorcycle": CityscapesClass("motorcycle", 32, 17, "vehicle", 7, True, False, (0, 0, 230)),
+            "bicycle": CityscapesClass("bicycle", 33, 18, "vehicle", 7, True, False, (119, 11, 32)),
+            "license plate": CityscapesClass("license plate", -1, -1, "vehicle", 7, False, True, (0, 0, 142)),
+        }
+    )
+
+    def _make_info(self) -> DatasetInfo:
+        name = "cityscapes"
+
+        return CityscapesDatasetInfo(
+            name,
+            categories=list(self.categories_to_details.keys()),
+            homepage="http://www.cityscapes-dataset.com/",
+            valid_options=dict(
+                split=("train", "val", "test", "train_extra"),
+                mode=("fine", "coarse"),
+            ),
+            extra=dict(categories_to_details=self.categories_to_details),
+        )
+
+    _FILES_CHECKSUMS = {
+        "gtCoarse.zip": "3555e09349ed49127053d940eaa66a87a79a175662b329c1a26a58d47e602b5b",
+        "gtFine_trainvaltest.zip": "40461a50097844f400fef147ecaf58b18fd99e14e4917fb7c3bf9c0d87d95884",
+        "leftImg8bit_trainextra.zip": "e41cc14c0c06aad051d52042465d9b8c22bacf6e4c93bb98de273ed7177b7133",
+        "leftImg8bit_trainvaltest.zip": "3ccff9ac1fa1d80a6a064407e589d747ed0657aac7dc495a4403ae1235a37525",
+    }
+
+    def resources(self, config: DatasetConfig) -> List[OnlineResource]:
+
+        if config.mode == "fine":
+            filenames = ("leftImg8bit_trainvaltest.zip", "gtFine_trainvaltest.zip")
+        else:
+            split_label = "trainextra" if config.split == "train_extra" else "trainvaltest"
+            filenames = (f"leftImg8bit_{split_label}.zip", "gtCoarse.zip")
+
+        resources: List[OnlineResource] = [
+            CityscapesResource(file_name=file_name, sha256=self._FILES_CHECKSUMS[file_name]) for file_name in filenames
+        ]
+        return resources
```

Comment:
Nit: I would prefer to avoid having an extra variable

```suggestion
        return [
            CityscapesResource(file_name=file_name, sha256=self._FILES_CHECKSUMS[file_name]) for file_name in filenames
        ]
```

If `mypy` complains we can cast

```suggestion
        return cast(List[OnlineResource], [
            CityscapesResource(file_name=file_name, sha256=self._FILES_CHECKSUMS[file_name]) for file_name in filenames
        ])
```

No strong opinion though.